[PodSecurity] Correct and clarify a few things (#29124)
kubernetes/website: content/en/docs/concepts/security/pod-security-admission.md

--feature-gates="...,PodSecurity=true"
```

## Pod Security levels

Pod Security admission places requirements on a Pod's [Security
Context](/docs/tasks/configure-pod-container/security-context/) and other related fields according
to the three levels defined by the [Pod Security Standards]: `privileged`, `baseline`, and

> `restricted`. Refer to the [Pod Security Standards] page for an in-depth look at those requirements.

Reviewer: I'd suggest we use inline links for the sake of link verification automation.

```suggestion
`restricted`. Refer to the [Pod Security Standards](/docs/concepts/security/pod-security-standards/)
page for an in-depth look at those requirements.
```

pod-security.kubernetes.io/<MODE>: <LEVEL>

# Optional: per-mode version label that can be used to pin the policy to the
# version that shipped with a given Kubernetes minor version (for example v{{< skew latestVersion >}}).
#
# MODE must be one of `enforce`, `audit`, or `warn`.
# VERSION must be a valid Kubernetes minor version, or `latest`.
pod-security.kubernetes.io/<MODE>-version: <VERSION>
```

Check out [Enforce Pod Security Standards with Namespace Labels](/docs/tasks/configure-pod-container/enforce-standards-namespace-labels) to see example usage.

## Workload resources and Pod templates

Pods are often created indirectly, by creating a [workload
object](https://kubernetes.io/docs/concepts/workloads/controllers/) such as a {{< glossary_tooltip
term_id="deployment" >}} or {{< glossary_tooltip term_id="job">}}. The workload object defines a

> _Pod template_ and a {{< glossary_tooltip term_id="controller" >}} for the workload resource creates

Reviewer 2: another very small nit:
```suggestion
_Pod template_ and a {{< glossary_tooltip term_id="controller" text="controller" >}} for the workload resource creates
```

We don't write “Controller” in title case because it's not a kind.

Pods based on that template. To help catch violations early, both the audit and warning modes are

> applied to the workload resources. However, enforce mode is **not** applied to [workload resources,

Reviewer 2: nit: (OK to merge now and fix this in a follow-up)
```suggestion
applied to the workload resources. However, enforce mode is **not** applied to workload resources,
```

# version that shipped with a given Kubernetes minor version (e.g. v{{< skew latestVersion >}}).
#
# MODE must be one of `enforce`, `audit`, or `warn`.
# VERSION must be a valid Kubernetes minor version, or `latest`.
pod-security.kubernetes.io/<MODE>-version: <VERSION>
```

Check out [Enforce Pod Security Standards with Namespace Labels](/docs/tasks/configure-pod-container/enforce-standards-namespace-labels) to see example usage.

## Pod Templates and Pod Controllers

Pods are often created indirectly, by creating a pod {{< glossary_tooltip term_id="controller" >}}

> object such as a {{< glossary_tooltip term_id="deployment" >}} or {{< glossary_tooltip term_id="job">}}.

Reviewer 3: +1 for "workload object" over "pod object"

To help catch violations early, both the audit and warning modes are applied to the pod
controller objects. However, enforce mode is **not** applied to pod controller objects, only to the

> resulting pod object.

Reviewer 3: ```suggestion
resulting pod objects.
```

> [pod controllers]: #pod-templates-and-pod-controllers

Reviewer: same above, pls use inline links when possible.
This flavor of links are difficult for the link-checker script to parse.

## Exemptions

You can define _exemptions_ from pod security enforcement in order allow the creation of pods that
would have otherwise been prohibited due to the policy associated with a given namespace.
Exemptions can be statically configured in the
[Admission Controller configuration](#configuring-the-admission-controller).

Exemptions must be explicitly enumerated. Requests meeting exemption criteria are _ignored_ by the

> Admission Controller (all `enforce`, `audit` and `warn` behaviors). Exemption dimensions include:

Reviewer 3: ```suggestion
Admission Controller (all `enforce`, `audit` and `warn` behaviors are skipped). Exemption dimensions include:
```

- **Usernames:** requests from users with an exempt authenticated (or impersonated) username are
  ignored.

> - **RuntimeClassNames:** pods and [pod controllers] specifying an exempt runtime class name are

Reviewer: here too.

  ignored.
- **Namespaces:** pods and [pod controllers] in an exempt namespace are ignored.

{{< caution >}}

> Most pods are created by a [pod controller], meaning that exempting an end user will only exempt

Reviewer: and here

Reviewer 3: ```suggestion
Most pods are created by a controller in response to a workload resource, meaning that exempting an end user will only exempt
```

> them from enforcement when creating pods directly, but not when creating a [pod controller] object.

Reviewer: we actually don't need to link every instance of the "pod controller", right?
